code/game/objects/items/bodybag.dm

```diff
@@ -89,4 +89,94 @@
 	// Make it possible to escape from bodybags in morgues and crematoriums
 	if(loc && (isturf(loc) || istype(loc, /obj/structure/morgue) || istype(loc, /obj/structure/crematorium)))
 		if(!open())
-			to_chat(user, "<span class='notice'>It won't budge!</span>")
\ No newline at end of file
+			to_chat(user, "<span class='notice'>It won't budge!</span>")
+
+/obj/item/bodybag/cryobag
+	name = "stasis bag"
+	desc = "A folded, non-reusable bag designed for the preservation of an occupant's body by stasis. Activates on occupant's unconscious state. Can be locked with medical ID."
+	icon = 'icons/obj/cryobag.dmi'
+	icon_state = "bodybag_folded"
+
+/obj/item/bodybag/cryobag/attack_self(mob/user)
+	var/obj/structure/closet/body_bag/cryobag/R = new /obj/structure/closet/body_bag/cryobag(user.loc)
+	R.add_fingerprint(user)
+	R.desc = desc
+	qdel(src)
+
+/obj/structure/closet/body_bag/cryobag
+	name = "stasis bag"
+	icon = 'icons/obj/cryobag.dmi'
+	item_path = /obj/item/bodybag/cryobag
+	var/used = 0
+	var/locked = 0
+	var/timer = 0
+	req_access = list(access_medical)
+
+/obj/structure/closet/body_bag/cryobag/Destroy()
+	processing_objects.Remove(src)
+	dump_contents()
+	return ..()
+
+/obj/structure/closet/body_bag/cryobag/dump_contents()
+	for(var/atom/movable/AM in src) //To prevent something standing on body_bag being affected
+		if(!ismob(AM))
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 0
+		used = 1	//We consider bag as used only if there is actual human in it
+	..()
+
+/obj/structure/closet/body_bag/cryobag/open()
+	. = ..()
+	if(used) //happens after dump_contents()
+		var/obj/item/O = new/obj/item(loc)
+		O.name = "used stasis bag"
+		O.icon = icon
+		O.icon_state = "bodybag_used"
+		O.desc = "Pretty useless now.."
+		qdel(src)
+
+
+/obj/structure/closet/body_bag/cryobag/close()
+	. = ..()
+	for(var/atom/movable/AM in src) //Atom already forcemoved into src in parent close() so we just searching from contents of src
+		if(!ismob(AM))
+			continue
+		var/mob/living/M = AM
+		M.in_stasis = 1
+		timer = world.time + STASIS_DELAY
+		processing_objects.Add(src)
+
+/obj/structure/closet/body_bag/cryobag/toggle(mob/user)
+	if(!opened && locked && !allowed(user))
+		to_chat(user, "<span class='warning'>Access denied. Medical access only.</span>")
+		return
+	..()
+
+/obj/structure/closet/body_bag/cryobag/process()
+	if(timer > world.time)
+		return
+	visible_message("<span class='notice'>[bicon(src)][name] run out of power. Stasis field disabled.</span>")

```

Comment:
bicon should be avoided where unnecessary. This classifies as unnecessary. This is also gramatically incorrect, change `[name] run out of power` to `[src] runs out of power.`